src/api.cpp

```diff
@@ -710,7 +710,9 @@ cl_int CLVK_API_CALL clGetDeviceInfo(cl_device_id dev,
         size_ret = sizeof(val_uint);
         break;
     case CL_DEVICE_IMAGE_MAX_BUFFER_SIZE:
-        val_sizet = device->vulkan_limits().maxImageDimension1D;
+        val_sizet =
+            std::min((uint64_t)device->vulkan_limits().maxTexelBufferElements,
+                     device->max_mem_alloc_size());
```

Comment:
Let's push this into a `cvk_device::image_max_buffer_size` function.

Reply:
Done